src/main/java/org/opensearch/ad/AnomalyDetectorJobRunner.java

```diff
@@ -649,7 +595,7 @@ private void indexAnomalyResultException(
             log.error("Failed to index anomaly result for " + detectorId, e);
         } finally {
             if (releaseLock) {
-                releaseLock(jobParameter, lockService, lock);
+                releaseLock(jobParameter, lock);
```

Comment:
Trying to understand. How are we handling the `lockService`?

Reply:
The `lockService` isn't directly available for extensions since extensions won't be installed within OpenSearch. To support acquiring/releasing a lock for an extension, we have exposed two REST APIs on job scheduler to acquire a lock (returns an object of type LockModel) and to release a lock (which updates the metadata lock index entry on job scheduler to allow a different job to run).

Reply:
@joshpalis  extensions maynot be installed within OpenSearch right now since we are working on running extensions out of process. But in near future, extensions can run in-proc within OpenSearch. We may want to think about this for near future.

Reply:
@minalsha Even if extensions run in-process, extensions are still initialized after the bootstrap of OpenSearch, which is when Job Scheduler class loads the plugin implementations of `ScheduledJobRunners`. This is when the `lockService` is provided for plugins. For extensions, we still woulnd't be able to access the lock service directly, that communication will have to be facilitated through the REST APIs that we exposed on Job Scheduler to acquire and release the lock

Reply:
sounds good. Thanks for explaining it.